improve readme for contributor experience

sebgoa opened the pull request:
**What this PR does / why we need it**:

This improves the top-level README. Removes broken links, removes links to make you bounce to another page, adds proper links to docs.
Shuffles things around a bit and improves reading.

@kubernetes/contributor-experience

<!-- Reviewable:start -->
---
This change is [<img src="https://reviewable.kubernetes.io/review_button.svg" height="34" align="absmiddle" alt="Reviewable"/>](https://reviewable.kubernetes.io/reviews/kubernetes/kubernetes/36817)
<!-- Reviewable:end -->

sebgoa committed:
improve readme for contributor experience

metral commented:
This PR removes links to external community efforts around k8s, e.g. Awesome-Kubernetes & CoreKube etc. - (I started CoreKube).

Is it worth it to place these efforts in a separate page dedicated for community resources as folks still find these sorts of efforts helpful?

sebgoa commented:
To plead my case, as I mentioned on twitter, I was doing a review of the README and when I hit the bottom I saw those links. CoreKube is really nice but the collections page is empty.

I pondered a lot on what to do with them, and then decided to remove them.

I think they would be better on the website. there is a big community page that needs lots of love. There is also a big Tools page that needs love as well.

metral commented:
Makes total sense 👍 .

I'll defer to Sarah to suggest a more appropriate home for these sorts of pages.

bgrant0607 commented:
cc @kubernetes/docs

bgrant0607 commented:
Copying from slack:

I don't think we should link to community resources from kubernetes/README.md. We should find a place on kubernetes.io for them. 

Files to remove:
* CONTRIB.md
* DESIGN.md
* contrib/README.md
* docs/warning.png
* docs/roadmap.md
* docs/troubleshooting.md
* docs/whatisk8s.md
* www/README.md

bgrant0607 commented on README.md:
```diff
@@ -2,22 +2,19 @@
+Kubernetes is an open source system for managing [containerized applications](http://kubernetes.io/docs/whatisk8s/) across multiple hosts,
+providing basic mechanisms for deployment, maintenance, and scaling of applications. Kubernetes is hosted by the Cloud Native Computing Foundation ([CNCF](https://www.cncf.io))
 
 Kubernetes is:
```
Please remove these bullets.

bgrant0607 commented on README.md:
```diff
@@ -26,138 +23,131 @@ Kubernetes is:
+
+<hr>
+
+### Kubernetes is ready for Production !
```
I'd just remove this subsection.

bgrant0607 commented on README.md:
```diff
@@ -26,138 +23,131 @@ Kubernetes is:
+
+Since the Kubernetes 1.0 release in July 2015 Kubernetes is ready for your production workloads. Check the [case studies](http://kubernetes.io/case-studies/).
+
+### Kubernetes can run anywhere !
```
I'd just remove this subsection.

bgrant0607 commented on README.md:
```diff
@@ -26,138 +23,131 @@ Kubernetes is:
+### Are you ...
+
+  * Interested in learning more about using Kubernetes?  Please see our user-facing documentation on [kubernetes.io](http://kubernetes.io). Try our [interactive tutorial](http://kubernetes.io/docs/tutorials/kubernetes-basics/) or take a free course on [Scalable Microservices with Kubernetes](https://www.udacity.com/course/scalable-microservices-with-kubernetes--ud615).
+  * Interested in hacking on the core Kubernetes code base, developing tools using the Kubernetes API or helping in anyway possible ?  Keep reading!
```
or helping with the project

bgrant0607 commented on README.md:
```diff
@@ -26,138 +23,131 @@ Kubernetes is:
+_As contributors and maintainers of this project, and in the interest of fostering an open and welcoming community, we pledge to respect all people who contribute through reporting issues, posting feature requests, updating documentation, submitting pull requests or patches, and other activities._
 
-## Concepts
+## Concepts Overview
```
Please remove this subsection.

bgrant0607 commented on README.md:
```diff
@@ -26,138 +23,131 @@ Kubernetes is:
+## Community
 
-Kubernetes documentation is organized into several categories.
+Do you want to help " shape the evolution of technologies that are container packaged, dynamically scheduled and microservices oriented? ". If you are a company, you should consider joining the [CNCF](https://cncf.io/about). For details about who's involved and how Kubernetes plays a role, read [the announcement](https://cncf.io/news/announcement/2015/07/new-cloud-native-computing-foundation-driv ...
```
container-packaged
microservices-oriented

English is right-associative.

bgrant0607 commented on README.md:
```diff
@@ -26,138 +23,131 @@ Kubernetes is:
-### Are you ready to add to the discussion?
-
-We have presence on:
+Join us on social media and read our blog:
```
I'd prefer to link to kubernetes.io (e.g., http://kubernetes.io/community/) and keep this simpler. If some of this info isn't easily found there, we should fix that.

bgrant0607 commented on README.md:
Hmm. I don't see info about kubecon on kubernetes.io. That would be useful to fix.

bgrant0607 commented on README.md:
@michelleN We'll need to update these links when the developer docs are moved.

bgrant0607 commented on README.md:
There's a link to the doc site above, so I think we can just remove this section.

bgrant0607 commented:
Thanks for the cleanup, @sebgoa!

michelleN commented on README.md:
Will do

sebgoa committed:
Additional cleanups and addressing review

sebgoa commented:
Ok addressed all comments.

I kept some links in the community section, just made it a bit more compact. I will make sure to revisit once the kubernetes.io/community page gets in a better state.

Ideally I would like to add a simple build section once I get passed https://github.com/kubernetes/kubernetes/issues/36655 , showing how you can build k8s in two bash commands will be powerful.

sebgoa commented:
I think we could also remove code-of-conduct.md since it is now a link to cncf. I am linking directly to it in the readme.

bgrant0607 commented:
@sebgoa code-of-conduct.md is currently part of the repo template. If we remove it here, we should make a similar change to the template. philips may have an opinion on that.

bgrant0607 commented:
Verify checks failed. 
https://k8s-gubernator.appspot.com/build/kubernetes-jenkins/pr-logs/pull/36817/pull-kubernetes-verify/4754/

bgrant0607 commented:
LGTM other than the check failures.

sebgoa committed:
fix munge-docs build errors

sebgoa commented:
ok tests are passing now. hadn't run munge-docs so fixed some broken links in docs/ docs/devel and examples/ folders.

bgrant0607 commented:
@saad-ali Would you mind if this were merged now?

saad-ali commented:
> @saad-ali Would you mind if this were merged now?

Doc changes are ok for post-code-freeze merge.

philips commented:
I am fine with the code-of-conduct move. I will do the same for the template project.
